File: app/src/main/java/com/unis/gameplatfrom/api/ApiException.java
package com.unis.gameplatfrom.api;

/**
 * Created by wulei on 16/5/27.
 */
@SuppressWarnings("DefaultFileTemplate")
public class ApiException extends RuntimeException {
    private int code;

    public ApiException(int code, String message) {
        super(message);
        this.code = code;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }
}

File: kotlin/src/main/java/com/unis/kotlin/api/ApiException.kt
package com.unis.kotlin.api

import java.lang.RuntimeException

class ApiException : RuntimeException{

    private var code: Int = 0

    constructor(code: Int, message: String): super(message){

        this.code = code
    }

    fun getCode(): Int {
        return code
    }

    fun setCode(code: Int) {
        this.code = code
    }

}